mlflow/projects/_project_spec.py

```diff
@@ -159,7 +159,7 @@ class Parameter(object):
     """A parameter in an MLproject entry point."""
     def __init__(self, name, yaml_obj):
         self.name = name
-        if isinstance(yaml_obj, str):
+        if isinstance(yaml_obj, six.string_types):
```

Comment:
Did you consider adding a utility `is_string_type` function?
When python 4 comes out, we prolly dont want to replace `six` with `twelve` all over. Also you can unit test that function with all sort of input -- numbers, unicode, bytes... etc